centrallix/osdrivers/objdrv_http.c

```diff
@@ -1712,10 +1719,21 @@ http_internal_GetPageStream(pHttpData inf)
 		inf->ContentType = nmSysStrdup(hdr_val);
 		if (strchr(inf->ContentType, ';'))
 		    *(strchr(inf->ContentType, ';')) = '\0';
+		/* add charset if present */
+		char * charsetStart = NULL;
+		if((charsetStart = strchr(hdr_val, "charset=")) != NULL && !inf->OverrideContentCharset)
+		    {
+		    inf->ContentCharset = nmSysStrdup(charsetStart+8);
+		    if (strchr(inf->ContentCharset, ';'))
+			*(strchr(inf->ContentCharset, ';')) = '\0';
+		    }
+		else if(inf->ExpectedContentCharset)
+		    inf->ContentCharset = nmSysStrdup(inf->ExpectedContentCharset);
 		}
 	    else
 		{
 		inf->ContentType = nmSysStrdup(inf->RestrictContentType);
+		inf->ContentCharset = nmSysStrdup(inf->ExpectedContentCharset);
```

Comment:
nmSysStrdup() will not accept a NULL input, which could happen if that property in the http object is unset.  Similar issue a few lines above.

Reply:
I mentioned before I thought I had tested it, but in order to get here there had to be no content type field in the response header, which was not a case I tested after all. 

I am not sure what is wrong with the one a few lines above; the other two usages of nmSysStrdup I added already had null checks?